crates/util/cometindex/src/indexer.rs

```diff
@@ -16,9 +15,9 @@ pub struct Indexer {
 }
 
 impl Indexer {
-    pub fn new() -> Self {
+    pub fn new(opts: Options) -> Self {
         Self {
-            opts: Options::parse(),
+            opts,
```

Comment:
I'm not sure this is desirable -- the original API was structured that way so that downstream consumers didn't have control over the CLI. Everyone gets the same binary, the only difference is which `AppView`s are hooked into it.

Reply:
If you want to hook in different AppViews you have to consumer pindexer as a library, in which case having control over your CLI seems much better to me, and doesn't cause odd spooky behavior.

Reply:
I don't fully understand the disagreement about exposing CLI opts: in the interest of making `pindexer` more accessible, e.g. via importing into other rust code, I'm biased toward making this change and dealing with the consequences later.

Reply:
Yes, this change should be rolled back, it's an explicit design goal to not allow control over the CLI, so that every Penumbra indexer has the same interface.